Instructions to compile `test2json` without local go (#226)
gotestyourself/gotestsum: docs/running-without-go.md

`gotestsum` may be run without Go as long as the package to be tested has
already been compiled using `go test -c`, and the `test2json` tool is available.

> The `test2json` tool can be compiled without installing go:

Reviewer: ```suggestion
The `test2json` tool can be compiled from the Go source tree so that it can be distributed to the environment that needs it.
```

```sh

> GOVERSION=1.13.5

Reviewer: ```suggestion
GOVERSION=1.17.6
```

We may as well update this to something recent while we are in here making changes.

OS=$(uname -s | sed 's/.*/\L&/')
mkdir -p gopath
GOPATH=$(realpath gopath)
HOME=$(realpath ./)
curl -L --silent https://go.dev/dl/go${GOVERSION}.${OS}-amd64.tar.gz | tar xz -C ./
env HOME=$HOME GOOS=linux GOARCH=amd64 CGO_ENABLED=0 GOPATH=$GOPATH ./go/bin/go build -o test2json -ldflags="-s -w" cmd/test2json

> rm -rf go
> rm -rf gopath

Reviewer: ```suggestion
```

Let's leave any cleanup out of these instructions so prevent anyone from accidentally deleting something they wanted to keep.
